File: next/security/user and permission/backend-ms1/src/main/java/ma/zs/stocky/ws/facade/admin/security/ModelPermissionRestAdmin.java
package  ma.zs.stocky.ws.facade.admin.security;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;

import ma.zs.stocky.bean.core.security.ModelPermission;
import ma.zs.stocky.dao.criteria.core.security.ModelPermissionCriteria;
import ma.zs.stocky.service.facade.admin.security.ModelPermissionAdminService;
import ma.zs.stocky.ws.converter.security.ModelPermissionConverter;
import ma.zs.stocky.ws.dto.security.ModelPermissionDto;
import ma.zs.stocky.zynerator.controller.AbstractController;
import ma.zs.stocky.zynerator.dto.AuditEntityDto;
import ma.zs.stocky.zynerator.util.PaginatedList;


import org.springframework.core.io.InputStreamResource;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import ma.zs.stocky.zynerator.process.Result;


import org.springframework.web.multipart.MultipartFile;
import ma.zs.stocky.zynerator.dto.FileTempDto;

@RestController
@RequestMapping("/api/admin/modelPermission/")
public class ModelPermissionRestAdmin  extends AbstractController<ModelPermission, ModelPermissionDto, ModelPermissionCriteria, ModelPermissionAdminService, ModelPermissionConverter> {



    @Operation(summary = "upload one modelPermission")
    @RequestMapping(value = "upload", method = RequestMethod.POST, consumes = "multipart/form-data")
    public ResponseEntity<FileTempDto> uploadFileAndGetChecksum(@RequestBody MultipartFile file) throws Exception {
        return super.uploadFileAndGetChecksum(file);
    }
    @Operation(summary = "upload multiple modelPermissions")
    @RequestMapping(value = "upload-multiple", method = RequestMethod.POST, consumes = "multipart/form-data")
    public ResponseEntity<List<FileTempDto>> uploadMultipleFileAndGetChecksum(@RequestBody MultipartFile[] files) throws Exception {
        return super.uploadMultipleFileAndGetChecksum(files);
    }

    @Operation(summary = "Finds a list of all modelPermissions")
    @GetMapping("")
    public ResponseEntity<List<ModelPermissionDto>> findAll() throws Exception {
        return super.findAll();
    }



    @Operation(summary = "Saves the specified  modelPermission")
    @PostMapping("")
    public ResponseEntity<ModelPermissionDto> save(@RequestBody ModelPermissionDto dto) throws Exception {
        return super.save(dto);
    }

    @Operation(summary = "Updates the specified  modelPermission")
    @PutMapping("")
    public ResponseEntity<ModelPermissionDto> update(@RequestBody ModelPermissionDto dto) throws Exception {
        return super.update(dto);
    }

    @Operation(summary = "Delete list of modelPermission")
    @PostMapping("multiple")
    public ResponseEntity<List<ModelPermissionDto>> delete(@RequestBody List<ModelPermissionDto> listToDelete) throws Exception {
        return super.delete(listToDelete);
    }
    @Operation(summary = "Delete the specified modelPermission")
    @DeleteMapping("")
    public ResponseEntity<ModelPermissionDto> delete(@RequestBody ModelPermissionDto dto) throws Exception {
            return super.delete(dto);
    }

    @Operation(summary = "Delete the specified modelPermission")
    @DeleteMapping("id/{id}")
    public ResponseEntity<Long> deleteById(@PathVariable Long id) throws Exception {
        return super.deleteById(id);
    }
    @Operation(summary = "Delete multiple modelPermissions by ids")
    @DeleteMapping("multiple/id")
    public ResponseEntity<List<Long>> deleteByIdIn(@RequestBody List<Long> ids) throws Exception {
            return super.deleteByIdIn(ids);
     }



    @Operation(summary = "Finds a modelPermission and associated list by id")
    @GetMapping("id/{id}")
    public ResponseEntity<ModelPermissionDto> findById(@PathVariable Long id) {
        return super.findById(id);
    }

    @Operation(summary = "Finds modelPermissions by criteria")
    @PostMapping("find-by-criteria")
    public ResponseEntity<List<ModelPermissionDto>> findByCriteria(@RequestBody ModelPermissionCriteria criteria) throws Exception {
        return super.findByCriteria(criteria);
    }

    @Operation(summary = "Finds paginated modelPermissions by criteria")
    @PostMapping("find-paginated-by-criteria")
    public ResponseEntity<PaginatedList> findPaginatedByCriteria(@RequestBody ModelPermissionCriteria criteria) throws Exception {
        return super.findPaginatedByCriteria(criteria);
    }

    @Operation(summary = "Exports modelPermissions by criteria")
    @PostMapping("export")
    public ResponseEntity<InputStreamResource> export(@RequestBody ModelPermissionCriteria criteria) throws Exception {
        return super.export(criteria);
    }

    @Operation(summary = "Gets modelPermission data size by criteria")
    @PostMapping("data-size-by-criteria")
    public ResponseEntity<Integer> getDataSize(@RequestBody ModelPermissionCriteria criteria) throws Exception {
        return super.getDataSize(criteria);
    }



    public ModelPermissionRestAdmin (ModelPermissionAdminService service, ModelPermissionConverter converter) {
        super(service, converter);
    }




}

File: nest-angular/security/backend-ms1/src/app/module/ws/admin/security/ModelPermissionRestAdmin.ts
import {Body, Controller,Delete , Get, Param, Post, Put} from '@nestjs/common';
import {ApiOperation, ApiTags} from '@nestjs/swagger';
import {PaginatedList} from "src/app/zynerator/util/PaginatedList";

import {ModelPermissionConverter} from "src/app/shared/converter/security/ModelPermissionConverter";
import {ModelPermissionDto} from "src/app/shared/dto/security/ModelPermissionDto";
import {ModelPermissionCriteria} from "src/app/shared/dao/criteria/core/security/ModelPermissionCriteria";
import {ModelPermissionAdminServiceImpl} from "src/app/module/service/admin/security/ModelPermissionAdminServiceImpl";


@ApiTags('Manages model permission services')
@Controller('api/admin/modelPermission')
export class ModelPermissionRestAdmin {

    constructor(private readonly service: ModelPermissionAdminServiceImpl,
                private readonly converter: ModelPermissionConverter) {
    }

    @ApiOperation({summary: 'Finds a list of all modelPermissions'})
    @Get()
    async findAll(): Promise<ModelPermissionDto[]> {
        const items = await this.service.findAll();
        return this.converter.toDtos(items);
    }

    @ApiOperation({summary: 'Finds a model permission by id'})
    @Get('id/:id')
    async findById(@Param('id') id: number): Promise<ModelPermissionDto> {
        const item = await this.service.findById(id);
        return this.converter.toDto(item);
    }

    @ApiOperation({summary: 'Deletes a model permission by id'})
    @Delete('id/:id')
    async deleteById(@Param('id') id: number): Promise<number> {
        return this.service.deleteById(id);
    }

    @ApiOperation({summary: 'Saves the specified model permission'})
    @Post()
    async save(@Body() dto: ModelPermissionDto): Promise<ModelPermissionDto> {
        const item = this.converter.toItem(dto);
        const savedItem = await this.service.save(item);
        return this.converter.toDto(savedItem);
    }


    @ApiOperation({summary: 'Updates the specified model permission'})
    @Put()
    async update(@Body() dto: ModelPermissionDto): Promise<ModelPermissionDto> {
        const item = this.converter.toItem(dto);
        const result = await this.service.updateWithAssociatedLists(item);
        return this.converter.toDto(result);
   }

    @ApiOperation({summary: 'Finds an optimized list of all modelPermissions'})
    @Get('optimized')
    async findAllOptimized(): Promise<ModelPermissionDto[]> {
        const result = await this.service.findAllOptimized();
        return result;
    }

    @ApiOperation({summary: 'Finds an optimized list of all modelPermissions'})
    @Post('find-by-criteria')
    async findByCriteria(@Body() criteria: ModelPermissionCriteria): Promise<ModelPermissionDto[]> {
        const items = await this.service.findByCriteria(criteria);
        return this.converter.toDtos(items);
    }

    @ApiOperation({summary: 'Finds an optimized list of all modelPermissions'})
    @Post('find-paginated-by-criteria')
    async findPaginatedByCriteria(@Body() criteria: ModelPermissionCriteria): Promise<PaginatedList<ModelPermissionDto>> {
        const paginated = await this.service.findPaginatedByCriteria(criteria);
        const dtos = this.converter.toDtos(paginated.list);
        return new PaginatedList<ModelPermissionDto>(dtos, paginated.dataSize);
    }



    @Post('multiple')
    async deleteMultiple(@Body() dtos: ModelPermissionDto[]): Promise<ModelPermissionDto[]> {
        const items = dtos.map(dto => this.converter.toItem(dto));
        const deletedItems = await this.service.deleteMultiple(items);
        return deletedItems.map(element => this.converter.toDto(element));
    }

}
